CODEOWNERS

```diff
@@ -0,0 +1,3 @@
+# release configuration
+/.release/                              @hashicorp/release-engineering
```

Comment:
Forgot to note that the release-engineering github team needs write access to the repo in order to be added as CODEOWNERS. I just made this change, so the error here should disappear.